tutorial/developing_packages/package_layout.rst

```diff
@@ -0,0 +1,333 @@
+  ``.``.
+- We set ``self.folders.build`` to be **./build/Release** or **./build/Debug** depending
+  on the ``build_type`` setting. These are the folders where we want the built binaries to
+  be located. 
+- The ``self.folders.generators`` folder is the location we set for all the files created
+  by the Conan generators. In this case, all the files generated by the ``CMakeToolchain``
+  generator will be stored there.
+
+.. note::
+
+    Please note that the values above are for a single-configuration CMake generator. To
+    support multi-configuration generators, such as Visual Studio, you should make some
+    changes to this layout. For a complete layout that supports both single-config and
+    multi-config, please check the :ref:`cmake_layout()<cmake_layout>` in the Conan
+    documentation.
+
+self.cpp
+--------
+
+This attribute is used to define **where consumers will find the package contents**
+(headers files, libraries, etc.) depending on wether the package is in editable mode or
+not.
+
+cpp.package
+^^^^^^^^^^^
+
+First, we set the information for `cpp.package`. This defines the contents of the package
+and its location relative to the folder where the package is stored in the local cache.
+Please note that defining this information is equivalent to defining `self.cpp_info` in
+the `package_info()` method. This is the information we defined:
+
+- ``self.cpp.package.libs``: we add the ``say`` library so that consumers know that they
+  should link with it. This is equivalent to declaring ``self.cpp_info.libs`` in the
+  ``package_info()`` method.
+
+- ``self.cpp.package.libdirs``: we add the ``lib`` folder so that consumers know that they
+  should should search there for the libraries. This is equivalent to declaring
+  ``self.cpp_info.libdirs`` in the ``package_info()`` method. Note that the default value
+  for ``libdirs`` in both the ``cpp_info`` and ``cpp.package`` is ``["lib"]`` so we could
+  have omitted that declaration.
+
+- ``self.cpp.package.includedirs``: we add the ``include`` folder so that consumers know
+  that they should should search there for the library headers. This is equivalent to
+  declaring ``self.cpp_info.includedirs`` in the ``package_info()`` method. Note that the
+  default value for ``includedirs`` in both the ``cpp_info`` and ``cpp.package`` is
+  ``["include"]`` so we could have omitted that declaration.
+
+To check how this information affects consumers we are going to do first do a ``conan create`` on
+the ``say`` package:
+
+.. code-block:: bash
+
+    $ cd say
+    $ conan create . -s build_type=Release
+
+When we call ``conan create``, Conan moves the recipe and sources declared in the recipe
+to be exported to the local Cache to a recipe folder and after that, it will create a
+separate package folder to build the binaries and store the actual package contents. If
+you check in the ``[YOUR_CONAN_HOME]/p`` folder, you will find two new folders similar to
```

Comment:
Maybe use ``conan cache path`` command?

Reply:
I thought so but then you have to find the package revision to have the complete reference and then using that to do something like `conan cache path say/1.0#rrev:pid#prev --folder build` that is not so straightforward

Reply:
``conan cache path`` is able to default to ``latest`` revisions as other ``list`` commands. ``conan cache path say/1.0`` would be ok

Reply:
But that gives you the latest recipe revision and in this case we more interested in the package revision, so you need at least a previous `conan list` command to extract the package-id

Reply:
The package_id (well, the folders too) are already in the output of the ``conan create``.
But yes, I agree, no need to mess here with this.

Reply:
I think I can add a note pointing to users that they can find that folder in the output, and also I could add like a tip or something pointing to the conan cache command but not giving the full instructions to not make it too long.